tensorboard/plugins/projector/projector_plugin_test.py

```diff
@@ -193,13 +227,39 @@ def testPluginIsActive(self):
     self._GenerateProjectorTestData()
     self._SetupWSGIApp()
 
-    # Embedding data is available.
+    # The is_active method makes use of a separate thread, so we mock threading
+    # behavior to make this test deterministic.
+    tf.test.mock.patch('threading.Thread', FakeThread).start()
```

Comment:
That works and removes the complexity of `FakeThread`. Wonderful idea.

1 alteration I made was replace `mock.assert_not_called()` with `self.assertEqual(0, mock.call_count)` because TF's version of mock lacked `assert_not_called`.